src/sentry/migrations/0216_fix_artifact_counts.py

```diff
@@ -0,0 +1,61 @@
+# Generated by Django 1.11.29 on 2021-06-21 07:28
+
+from django.db import migrations
+from django.db.models import Q
+
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+#: Copied from releasefile.py
+ARTIFACT_INDEX_FILENAME = "artifact-index.json"
+
+
+def uncount_internal_releasefiles(apps, schema_editor):
+    # artifact_count=None implicitly means "count as 1".
+    # Internal release files (artifact index, release archives) created *before*
+    # 0211_add_artifact_count must not be counted as 1 but rather as 0, because their
+    # individual files were saved with a count of None as well.
+    ReleaseFile = apps.get_model("sentry", "ReleaseFile")
+    db_alias = schema_editor.connection.alias
+    qs = ReleaseFile.objects.using(db_alias).filter(
+        Q(name=ARTIFACT_INDEX_FILENAME)
+        | (Q(name__startswith="release-artifacts") & Q(name__endswith=".zip")),
+        artifact_count=None,
+    )
+    for releasefile in RangeQuerySetWrapperWithProgressBar(qs, step=1000):
```

Comment:
`RangeQuerySetWrapperWithProgressBar` should be used in conjunction with a performant query (ex: `objects.all()` and filtering of items can be done in-memory.